pkg/runtest/run.go

```diff
@@ -21,69 +22,75 @@ import (
 	"sort"
 	"strconv"
 	"strings"
+	"sync"
 
 	"github.com/google/syzkaller/pkg/csource"
 	"github.com/google/syzkaller/pkg/flatrpc"
+	"github.com/google/syzkaller/pkg/fuzzer/queue"
 	"github.com/google/syzkaller/pkg/ipc"
 	"github.com/google/syzkaller/prog"
 	"github.com/google/syzkaller/sys/targets"
+	"golang.org/x/sync/errgroup"
 )
 
-type RunRequest struct {
-	Bin    string
-	P      *prog.Prog
-	Opts   ipc.ExecOpts
-	Repeat int
+type runRequest struct {
+	*queue.Request
 
-	Done   chan struct{}
-	Output []byte
-	Info   ipc.ProgInfo
-	Err    error
+	err     error
+	result  *queue.Result
+	results *ipc.ProgInfo // the expected results
 
-	finished chan struct{}
-	results  *ipc.ProgInfo
-	name     string
-	broken   string
-	skip     string
+	name   string
+	broken string
+	skip   string
 }
 
 type Context struct {
 	Dir          string
 	Target       *prog.Target
 	Features     flatrpc.Feature
 	EnabledCalls map[string]map[*prog.Syscall]bool
-	Requests     chan *RunRequest
 	LogFunc      func(text string)
 	Retries      int // max number of test retries to deal with flaky tests
 	Verbose      bool
 	Debug        bool
 	Tests        string // prefix to match test file names
+
+	orderLogs sync.Mutex
+	executor  queue.PlainQueue
 }
 
 func (ctx *Context) log(msg string, args ...interface{}) {
+	ctx.orderLogs.Lock()
```

Comment:
What happens w/o this mutex? Any reasonable logging should serialize output on its own, otherwise any log call anywhere in the program would somehow need to use a global mutex.